### PR TITLE
Split integration tests into their own folder

### LINKED ISSUE #25: controller.kubernetes.io/queue-name annotation not registered
The code in this repo uses an annotation, `controller.kubernetes.io/queue-name`, that is not registered in https://kubernetes.io/docs/reference/labels-annotations-taints/

We should either:
- register and document the annotation
- avoid specifying `controller.kubernetes.io` as the namespace for that annotation, and instead require specifying it as a command line option to the app. That way, end-users wouldn't assume that any particular namespace is expected.
- use another namespace, that is appropriate for kueue.

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

- Move integration tests into their own folder so that we can more easily run unit tests.
- Put controllers into pkg, splitting them in two kinds: core and workload (intended for integrating diverse custom job APIs).

#### Which issue(s) this PR fixes:

Part of #20
Fixes #25

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -56,8 +56,12 @@ vet: ## Run go vet against code.
...
 
 .PHONY: test
-test: manifests generate fmt vet envtest ## Run tests.
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out
+test: generate fmt vet ## Run tests.
+	go test ./pkg/... -coverprofile cover.out
+
+.PHONY: test-integration
+test-integration: manifests generate fmt vet envtest ## Run tests.
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./test/integration/... -coverprofile cover.out
```
Comment (ahg-g) [inline Makefile:64]: do we need coverage profile for integration?

Comment (alculquicondor) [reply Makefile]: Removed

**Pull request merged**